tests/resources/feature/test_feature.py

```diff
@@ -0,0 +1,103 @@
+#  Copyright 2023 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from unittest.mock import Mock
+
+import pytest
+
+from hathor.feature_activation.feature import Feature
+from hathor.feature_activation.feature_service import FeatureService
+from hathor.feature_activation.model.criteria import Criteria
+from hathor.feature_activation.model.feature_state import FeatureState
+from hathor.feature_activation.resources.feature import FeatureResource
+from hathor.feature_activation.settings import Settings
+from hathor.transaction import Block
+from hathor.transaction.storage import TransactionStorage
+from tests.resources.base_resource import StubSite
+
+
+@pytest.fixture
+def web():
+    best_block = Mock(spec_set=Block)
+    best_block.get_feature_activation_bit_counts = Mock(return_value=[0, 1, 0, 0])
+
+    tx_storage = Mock(spec_set=TransactionStorage)
+    tx_storage.get_best_block = Mock(return_value=best_block)
+
+    def get_state(*, block: Block, feature: Feature) -> FeatureState:
+        return FeatureState.ACTIVE if feature is Feature.NOP_FEATURE_1 else FeatureState.STARTED
+
+    feature_service = Mock(spec_set=FeatureService)
+    feature_service.get_state = Mock(side_effect=get_state)
+
+    settings = Settings(
```

Comment:
I think using simply `settings` might be very misleading and error prone. Maybe rename it to `feature_settings`?

Reply:
Done in https://github.com/HathorNetwork/hathor-core/pull/634/commits/6a8d8ab3d82cbd99c98536abb421644cab724364